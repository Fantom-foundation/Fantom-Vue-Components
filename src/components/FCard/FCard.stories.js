import FCard from './FCard.vue';

export default {
    title: 'FCard',
    component: FCard,
};

export const Default = () => ({
    components: { FCard },
    template: `
        <div>
            <f-card></f-card>
        </div>
    `,
});

export const DefaultSlot = () => ({
    components: { FCard },
    template: `
        <div>
            <f-card>
                <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Accusantium deserunt doloribus eius esse eum excepturi, inventore ipsam iure laboriosam nam natus nobis non nostrum numquam placeat quos rerum tempora totam!</p>
                <p>A consequuntur corporis deleniti dolore dolorem ducimus eius, hic in ipsa iure laboriosam magnam natus obcaecati, placeat possimus provident quasi quos repudiandae sequi tenetur! Alias amet deleniti mollitia odit unde?</p>
                <p>Amet consequuntur cumque deleniti dolores et eveniet excepturi facilis fugiat id laudantium magni nam nemo, nihil, quos repellendus sapiente soluta suscipit temporibus ullam ut. Cum expedita illum minus numquam rem!</p>
                <p>Error facere, mollitia natus nemo nobis porro recusandae sit temporibus velit? Amet assumenda earum esse et facere, ipsam ipsum laboriosam nam, nesciunt nihil non nostrum omnis perspiciatis rem repellendus repudiandae.</p>
            </f-card>
        </div>
    `,
});
