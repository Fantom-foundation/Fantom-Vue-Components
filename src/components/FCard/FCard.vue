<template>
    <div class="fcard">
        <slot></slot>
    </div>
</template>

<script>
export default {
    name: 'FCard',
};
</script>

<style lang="scss">
@use "style";
</style>
