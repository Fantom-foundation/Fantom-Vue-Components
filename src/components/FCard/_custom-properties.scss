.fcard {
    --fcard-padding: var(--f-spacer-5);
    --fcard-mw-padding: var(--f-spacer-3);
    --fcard-background: #fff;
    --fcard-border-radius: var(--f-border-radius-3);
    --fcard-box-shadow: 0 8px 16px 0 rgba(48,49,51,0.1),0 0 1px 0 rgba(48,49,51,0.05);
}
