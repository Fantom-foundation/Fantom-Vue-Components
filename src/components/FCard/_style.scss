@use "custom-properties";

.fcard {
    padding: var(--fcard-padding);
    background: var(--fcard-background);
    border-radius: var(--fcard-border-radius);
    box-shadow: var(--fcard-box-shadow);

    > h1, > .h1, > h2, > .h2, > h3, > .h3 {
        margin-top: 0;
    }
}
