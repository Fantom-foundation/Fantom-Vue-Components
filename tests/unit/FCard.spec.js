import { mount } from '@vue/test-utils';
import FCard from '@/components/FCard/FCard.vue';

let wrapper = null;

function destroyWrapper() {
    if (wrapper) {
        wrapper.destroy();
    }
}

function createWrapper({ propsData = {}, slots = {} } = {}) {
    destroyWrapper();

    wrapper = mount(FCard, {
        slots,
        propsData,
    });
}

describe('FCard', () => {
    it('should render correctly', () => {
        createWrapper();

        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render content in default slot', () => {
        createWrapper({
            slots: {
                default: ['<span>content</span>'],
            },
        });

        expect(wrapper.html()).toContain('<span>content</span>');
    });
});
