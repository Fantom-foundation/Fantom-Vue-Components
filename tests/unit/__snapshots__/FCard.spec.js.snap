// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FCard should render correctly 1`] = `
<div
  class="fcard"
/>
`;
